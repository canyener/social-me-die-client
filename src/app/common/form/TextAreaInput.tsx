import { FieldRenderProps } from "react-final-form"
import { Form, FormFieldProps, Label } from "semantic-ui-react"

interface IProps
  extends FieldRenderProps<string, HTMLTextAreaElement>,
  FormFieldProps { }

const TextAreaInput: React.FC<IProps> = ({
  input,
  width,
  rows,
  placeholder,
  meta: { touched, error }
}) => {
  return (
    <Form.Field error={touched && !!error} width={width}>
      <textarea rows={rows} {...input} placeholder={placeholder} /> 
      {
        touched && error && (
          <Label basic color='red'>
            {error}
          </Label>
        )
      }
    </Form.Field>
  )
}

export default TextAreaInput