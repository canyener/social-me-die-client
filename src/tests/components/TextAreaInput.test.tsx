import { shallow } from 'enzyme'
import TextAreaInput from '../../app/common/form/TextAreaInput'

test('Should render TextAreaInput correctly without any given errors', () => {
  const wrapper = shallow(<TextAreaInput
    width='11'
    placeholder='Test TextArea Placeholder'
    rows={3}
    input={{
      onBlur: jest.fn(),
      onChange: jest.fn(),
      onFocus: jest.fn(),
      name: 'Test textarea name',
      value: 'Test textarea value'
    }}
    meta={{
      touched: false,
      error: undefined
    }}
  />)

  expect(wrapper).toMatchSnapshot()
})

test('Should set Form.Field error to true and render Label with given error message', () => {
  const wrapper = shallow(<TextAreaInput
    width='11'
    placeholder='Test TextAreaPlaceholder'
    rows={3}
    input={{
      onBlur: jest.fn(),
      onChange: jest.fn(),
      onFocus: jest.fn(),
      name: 'Test textarea name',
      value: 'Test textarea value'
    }}
    meta={{
      touched: true,
      error: 'test error message'
    }}
  />)

  expect(wrapper).toMatchSnapshot()
})