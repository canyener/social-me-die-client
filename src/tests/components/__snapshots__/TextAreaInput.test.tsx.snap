// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render TextAreaInput correctly without any given errors 1`] = `
<FormField
  error={false}
  width="11"
>
  <textarea
    name="Test textarea name"
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    onFocus={[MockFunction]}
    placeholder="Test TextArea Placeholder"
    rows={3}
    value="Test textarea value"
  />
</FormField>
`;

exports[`Should set Form.Field error to true and render Label with given error message 1`] = `
<FormField
  error={true}
  width="11"
>
  <textarea
    name="Test textarea name"
    onBlur={[MockFunction]}
    onChange={[MockFunction]}
    onFocus={[MockFunction]}
    placeholder="Test TextAreaPlaceholder"
    rows={3}
    value="Test textarea value"
  />
  <Label
    basic={true}
    color="red"
  >
    test error message
  </Label>
</FormField>
`;
